src/ape/managers/chain.py

```diff
@@ -377,89 +383,109 @@ def _get_block(self, block_id: BlockID) -> BlockAPI:
         return self.provider.get_block(block_id)
 
 
-class AccountHistory(BaseManager):
+class AccountHistory(BaseInterfaceModel):
     """
     A container mapping account addresses to the transaction from the active session.
     """
 
-    _address_list_map: Dict[AddressType, List[ReceiptAPI]] = {}
-
-    # Facilitates faster look-ups.
-    _hash_to_receipt_map: Dict[str, ReceiptAPI] = {}
+    address: AddressType
+    _transaction_list: List[ReceiptAPI] = []
 
-    def __getitem__(self, address: Union[BaseAddress, AddressType, str]) -> List[ReceiptAPI]:
+    @property
+    def outgoing(self) -> Iterator[ReceiptAPI]:
         """
-        Get the list of transactions from the active session for the given address.
-
-        Args:
-            address (``AddressType``): The sender of the desired transactions.
-
-        Returns:
-            List[:class:`~ape.api.transactions.TransactionAPI`]: The list of transactions. If there
-            are no recorded transactions, returns an empty list.
+        All outgoing transactions.
         """
-
-        address_key: AddressType = self.conversion_manager.convert(address, AddressType)
         explorer = self.provider.network.explorer
         explorer_receipts = (
-            [r for r in explorer.get_account_transactions(address_key)] if explorer else []
+            [r for r in explorer.get_account_transactions(self.address)] if explorer else []
         )
         for receipt in explorer_receipts:
+            # TODO: Use cache
+            if receipt.txn_hash not in [t.txn_hash for t in self._transaction_list]:
+                yield receipt
 
-            # NOTE: Cache the receipt by its sender and not by the address sent to the explorer API
-            #  because explorers return various receipts when given contract addresses.
-            sender: AddressType = receipt.sender
-            if receipt.txn_hash not in [r.txn_hash for r in self._address_list_map.get(sender, [])]:
-                self.append(receipt)
+    @property
+    def all(self) -> Iterator[ReceiptAPI]:
+        """
+        All account transactions.
+        """
+        yield from self.outgoing
+        yield from self._transaction_list
```

Comment:
Might be good to yield "in order" meaning pull from each iterator whatever the oldest timestamp is (or newest, just document what direction it goes in and be consistent)